lib/common/MongoDbTransactionStore.ts

```diff
@@ -151,6 +151,14 @@ export default class MongoDbTransactionStore implements ITransactionStore {
     await this.transactionCollection!.deleteMany({ transactionNumber: { $gt: Long.fromNumber(transactionNumber) } });
   }
 
+  /**
+   * Remove transactions by transaction time hash
+   * @param transactionTimeHash the transaction time hash which the transactions should be removed for
+   */
+  public async removeTransactionByTransactionTimeHash (transactionTimeHash: string) {
+    await this.transactionCollection!.deleteMany({ transactionTimeHash: { $eq:  transactionTimeHash } });
```

Comment:
I wonder if it is worthwhile/overkill to index by `transactionTimeHash` now? It is obviously not a problem now, but I would imagine multiple passes against millions of transactions is going to take a while.